lib/db_ido/dbobject.cpp

```diff
@@ -119,26 +119,17 @@ void DbObject::SendConfigUpdateHeavy(const Dictionary::Ptr& configFields)
 
 	ASSERT(configFields->Contains("config_hash"));
 
-	ConfigObject::Ptr object = GetObject();
+	Dictionary::Ptr configHashField = new Dictionary({{"config_hash", configFields->Get("config_hash")}});
+	configFields->Set("config_hash", Empty);
 
-	DbQuery query;
-	query.Table = GetType()->GetTable() + "s";
-	query.Type = DbQueryInsert | DbQueryUpdate;
-	query.Category = DbCatConfig;
-	query.Fields = configFields;
-	query.Fields->Set(GetType()->GetIDColumn(), object);
-	query.Fields->Set("instance_id", 0); /* DbConnection class fills in real ID */
-	query.Fields->Set("config_type", 1);
-	query.WhereCriteria = new Dictionary({
-		{ GetType()->GetIDColumn(), object }
-	});
-	query.Object = this;
-	query.ConfigUpdate = true;
-	OnQuery(query);
+	// Since all sub-tables are relying on the inserted parent ID, we have to firstly insert/update
+	// the entries of the parent table without the config_hash column.
+	UpsertConfigFields(configFields);
 
 	m_LastConfigUpdate = Utility::GetTime();
 
-	OnConfigUpdateHeavy();
+	// Trigger config heavy udpates of the child classes and lastly update the config_hash column of the current type.
+	OnConfigUpdateHeavy(configHashField);
```

Comment:
... at the end of this method?

Reply:
The config_hash has to be updated before this nonsense happens, so yes, I've to call that parent method if it's overridden.
https://github.com/Icinga/icinga2/blob/3d64240ee36af577e2cafa467132f3d622e24551/lib/db_ido/hostdbobject.cpp#L334
https://github.com/Icinga/icinga2/blob/3d64240ee36af577e2cafa467132f3d622e24551/lib/db_ido/hostdbobject.cpp#L342-L349